components/style/properties/properties.mako.rs

```diff
@@ -1328,10 +1329,26 @@ impl ToCss for PropertyDeclaration {
 }
 
 impl MallocSizeOf for PropertyDeclaration {
-    fn malloc_size_of_children(&self, _malloc_size_of: MallocSizeOfFn) -> usize {
-        // The variants of PropertyDeclaration mostly (entirely?) contain
-        // scalars, so this is reasonable.
-        0
+    fn malloc_size_of_children(&self, malloc_size_of: MallocSizeOfFn) -> usize {
+        match *self {
+            % for property in data.longhands:
+                % if property.boxed and property.is_vector:
+                    (this should not happen! it is not smart to box a vector here)
+                % elif property.boxed:
+                    PropertyDeclaration::${property.camel_case}(ref sv_box) => {
+                        sv_box.malloc_shallow_size_of_box(malloc_size_of)
+                    }
+                % elif property.is_vector:
+                    PropertyDeclaration::${property.camel_case}(ref sv_vec) => {
+                        sv_vec.0.malloc_shallow_size_of_vec(malloc_size_of)
+                    }
+                % endif
+            % endfor
+            PropertyDeclaration::CSSWideKeyword(..) => 0,
+            PropertyDeclaration::WithVariables(..) => 0,
```

Comment:
Variables contain strings and so on, fwiw (though they're uncommon).

Reply:
I'm focusing on the big things at the moment, using DMD data to drive where the reporters are added.